src/Essentials/src/Permissions/Permissions.android.cs

```diff
@@ -357,6 +326,44 @@ public partial class Microphone : BasePlatformPermission
 				new (string, bool)[] { (Manifest.Permission.RecordAudio, true) };
 		}
 
+		public partial class NearbyDevices : BasePlatformPermission
+		{
+			public override (string androidPermission, bool isRuntime)[] RequiredPermissions
+			{
+				get
+				{
+					var permissions = new List<(string, bool)>();
+					// When targeting Android 11 or lower, AccessFineLocation is required for Bluetooth.
+					// For Android 12 and above, it is optional.
+					if (Application.Context.ApplicationInfo.TargetSdkVersion <= BuildVersionCodes.R || IsDeclaredInManifest(Manifest.Permission.AccessFineLocation))
+						permissions.Add((Manifest.Permission.AccessFineLocation, true));
+
+#if __ANDROID_31__
+					if (OperatingSystem.IsAndroidVersionAtLeast(31) && Application.Context.ApplicationInfo.TargetSdkVersion >= BuildVersionCodes.S)
+					{
+						// new runtime permissions on Android 12
+						if (IsDeclaredInManifest(Manifest.Permission.BluetoothScan))
+							permissions.Add((Manifest.Permission.BluetoothScan, true));
+						if (IsDeclaredInManifest(Manifest.Permission.BluetoothConnect))
+							permissions.Add((Manifest.Permission.BluetoothConnect, true));
+						if (IsDeclaredInManifest(Manifest.Permission.BluetoothAdvertise))
+							permissions.Add((Manifest.Permission.BluetoothAdvertise, true));
+					}
+#endif
+#if __ANDROID_33__
+					if (OperatingSystem.IsAndroidVersionAtLeast(33) && Application.Context.ApplicationInfo.TargetSdkVersion >= BuildVersionCodes.Tiramisu)
+					{
+						// new runtime permission on Android 13
+						if (IsDeclaredInManifest(Manifest.Permission.NearbyWifiDevices))
+							permissions.Add((Manifest.Permission.NearbyWifiDevices, true));
+					}
+#endif
+
+					return permissions.ToArray();
+				}
+			}
+		}
```

Comment:
I would highly consider splitting this into two. If I were to request this and have both Nearby WiFi and Nearby BLE device discovery, wouldn't I see two permission dialogs?

With this approach I cannot request one or the other without seeing both. Personally I would prefer just having `NearbyWifiDevices` as a separate permission.

Reply:
Maybe I am mistaken, I guess it is easy to test, just adding both permissions and see:

```xml
<uses-permission android:name="android.permission.NEARBY_WIFI_DEVICES" android:usesPermissionFlags="neverForLocation" />
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" android:usesPermissionFlags="neverForLocation" />
```

Reply:
> I would highly consider splitting this into two. If I were to request this and have both Nearby WiFi and Nearby BLE device discovery, wouldn't I see two permission dialogs?

No, you don't. IIUC the permission is granted for the whole group ("Nearby devices"). So if you first request Bluetooth permissions and WiFi afterwards, you only see a single dialog for the BT request. If this is granted, no second dialog for WiFi is shown.

> With this approach I cannot request one or the other without seeing both.

Not true. Which permissions exactly you request is determined by the manifest. If you only request BT, you'll see a request that mentions BT. If you only request WiFi, you'll see a request that mentions "nearby WiFi devices". I haven't checked how the dialog looks if you request both at the same time.

> Personally I would prefer just having `NearbyWifiDevices` as a separate permission.

That's what I started with, but @mattleibow had the opposite preference, so I changed it. Personally I don't care all that much, but it would be really nice if the WiFi permissions would make it into .NET 8.

Reply:
Alright, thanks for clearing that up :)